merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/MissionModelProcessor.java

```diff
@@ -80,22 +82,48 @@ public boolean process(final Set<? extends TypeElement> annotations, final Round
     for (final var element : roundEnv.getElementsAnnotatedWith(MissionModel.class)) {
       final var packageElement = (PackageElement) element;
       try {
-        final var missionModelRecord = missionModelParser.parseMissionModel(packageElement);
+        final var missionModelRecord$ = missionModelParser.parseMissionModel(packageElement);
 
         final var generatedFiles = new ArrayList<>(List.of(
-            missionModelGen.generateMerlinPlugin(missionModelRecord),
-            missionModelGen.generateSchedulerPlugin(missionModelRecord)));
+            missionModelGen.generateMerlinPlugin(missionModelRecord$),
+            missionModelGen.generateSchedulerPlugin(missionModelRecord$)));
 
-        missionModelRecord.modelConfigurationType
-            .flatMap(configType -> missionModelGen.generateMissionModelConfigurationMapper(missionModelRecord, configType))
+        missionModelRecord$.modelConfigurationType
+            .flatMap(configType -> missionModelGen.generateMissionModelConfigurationMapper(missionModelRecord$, configType))
             .ifPresent(generatedFiles::add);
 
         generatedFiles.addAll(List.of(
-            missionModelGen.generateMissionModelFactory(missionModelRecord),
-            missionModelGen.generateSchedulerModel(missionModelRecord),
-            missionModelGen.generateActivityActions(missionModelRecord),
-            missionModelGen.generateActivityTypes(missionModelRecord)
+            missionModelGen.generateMissionModelFactory(missionModelRecord$),
+            missionModelGen.generateSchedulerModel(missionModelRecord$),
+            missionModelGen.generateActivityActions(missionModelRecord$),
+            missionModelGen.generateActivityTypes(missionModelRecord$)
         ));
+
+        final var autoValueMapperRequests = roundEnv.getElementsAnnotatedWith(AutoValueMapper.class).stream().map(Element::asType).toList();
+        {
+          final var recordType = elementUtils.getTypeElement("java.lang.Record").asType();
+          for (final var request : autoValueMapperRequests) {
+            if (!typeUtils.isSubtype(request, recordType)) {
+              messager.printError("AutoValueMapper is only allowed on record types. %s is not a record".formatted(request.toString()));
```

Comment:
Where possible, interpolate type names from class references. If the type ever gets renamed, we want to make sure it gets renamed in this string, too.
```suggestion
              messager.printError("@%s is only allowed on record types. %s is not a record".formatted(AutoValueMapper.class.getSimpleName(), request.toString()));
```
(It's also nice for users if we explicitly prefix annotation type names with `@`.)

Reply:
Good point, change made :+1: